### ISSUE TITLE
waf compilation on mac after the removal of Homebrew

### ISSUE THREAD

**Issue by fre123ber** [mpv-player/mpv#7046]
Platform: macOS Catalina 10.15

Hello, like many people I use mpv every day as the fastest, most accurate and up-to-date video player in the world. Because I don't know almost anything about software development, it took a bit for me to figure out how to use homebrew to have the latest HEAD version. I have tried to build mpv with homebrew but the formula has been recently removed, I guess because of the continuous bugs related to ffmpeg/Libav.
I read on mpv's project page on GitHub that you now have to compile mpv with waf. Would someone be so kind as to make a guide for beginners on how to compile mpv with waf?

Also, I feel that for an important open-source project such as this there should be a unified platform/blog to inform everyone of new features and changes. Maybe a reddit page? It would be nice if mpv tried to be a little bit more user-friendly.
The latest release dates back to last year with 826 commits ahead as of now. Why not a release in such a long time? Is there any coordination between the project developers?
Thank you a lot in advance for the answers.

**Comment by wm4** [comment]
Have you tried to read the README? https://github.com/mpv-player/mpv#compilation

Homebrew AFAIK removed the package because Apple broke compatibility out of incompetence.

**Comment by wm4** [comment]
Oh, I also have to say that trying to build this yourself will probably result only in pain. There's a thing called homebrew cask (see mpv.io) which you should probably use instead.

**Comment by fre123ber** [comment]
I know, homebrew cask just installs the latest build from laboratory.stolendata.net/~djinn/mpv_osx/.
I don't see who mpv is for if almost anyone that compiles it is an "expert" and the latest release is not official and doesn't even support dark mode - as an example of a feature, not that the UI isn't dark or that it crashes on macOS, but system support for it simply was added later, along with swift changes, codecs, formats, libraries etc...

**Comment by wm4** [comment]
Oh, I thought homebrew could build stuff itself. There is mpv-build, but it's quite Linux centered, especially the instructions: https://github.com/mpv-player/mpv-build/

**Comment by Akemi** [comment]
cask unfortunately is just distributing binaries/bundles. tbh, i felt like the homebrew guys were just looking for a reason to remove it. there were various things they could have done, but they didn't want to.

anyway you can just [download](https://github.com/Homebrew/homebrew-core/blob/03ac5ca999877ff8e522a354aaeb246815f715b5/Formula/mpv.rb) the last mpv formula and install it like this:

```
brew install --HEAD /PATH/TO/DOWNLOADED/mpv.rb
```

**Comment by avih** [comment]
Is it possible to grab binary artifacts from CI builds? would they be useful enough for users who want to try the bleeding edge mpv?

**Comment by wm4** [comment]
Probably yes.. but then you'd need to package them with the required dependencies. And for "usable" builds you may have to code sign them, which will require you to pay for a certificate.

**Comment by avih** [comment]
For future reference, if one trusts a binary, they can sign it locally. If it's replaced, e.g. with a newer version, then it needs to be signed again.

TL;DR:
Sign a bundle:
- `sudo codesign --force --deep --sign - path/to/bundle.app`
Sign a single executable: 
- `sudo codesign --force        --sign - path/to/executable`

Reference: https://apple.stackexchange.com/a/150711

**Comment by Akemi** [comment]
> Probably yes.. but then you'd need to package them with the required dependencies. And for "usable" builds you may have to code sign them, which will require you to pay for a certificate.

the signing part isn't really the case for everything up to 10.14.x. if that's about 10.15 i couldn't test that yet though.

packinging shouldn't be a problem with the osx-bundle script. building for older platforms as target might be.
